### ISSUE TITLE
Server display names don't work

### ISSUE THREAD

**Issue by deesoff** [estroBiologist/pluralchum#4]
The plugin seems to always overwrite server specific display names with whatever that member's base display name is.

**Comment by soulslabsys** [comment]
This issue also occurs with server tags, including disabling your tag in a server. Example of this issue below:

**Expected behaviour when setting server display names or tags (example used is disabling the tag in a server):** 
![a](https://user-images.githubusercontent.com/73873252/155825074-47db88ab-641f-4db8-ace5-72e8350bd574.png)
Server tag doesn't display, as per the server's settings

**What actually happens when the plugin is enabled:**
![b](https://user-images.githubusercontent.com/73873252/155825091-499fa05b-43b0-4919-8bbd-210387aed997.png)
Server tag still displays, despite it being disabled for that server

**Comment by transdisaster** [comment]
Aware of this, and working on it. The difficulty is that getting this info properly requires both authenticating and doing follow-up requests with the API, which is both a performance hit, and a security responsibility I'm not really qualified to handle. As of right now, the plugin uses only public information from the API, and I'd like to keep it that way if possible. 

I appreciate the report, though, and I'll hopefully at least have a solution for servernames in the next release.

**Comment by spiralw** [comment]
right now the servername API is authentication locked since I don't want to leak "this user is in this server". it might be possible in the future to log in to the pluralkit API with OAuth2 so an API user can read other people's server settings if they share a server - would that work for you?

(love this plugin, btw!)

**Comment by sylveon-irl** [comment]
server system tags also do not work. it would be wonderful if we could get a string hack for that as well, though it's understandable if you feel that time is better spent elsewhere. thank you for all your hard work!

**Comment by estroBiologist** [comment]
i do plan to fix these, it's just that right now the plugin doesn't have enough data to figure out what the servername and servertag is reliably. i'll need to make the plugin authenticate itself with PK's web api to do that, which is a security headache-and-a-half. it might take a while, but they're at the top of my to-do list!

**Comment by estroBiologist** [comment]
also, about a potential similar hack for servertags: right now, the servername hack works on the assumption that there's no special servertag set. it uses the default system tag to try to deduce what the servername is from the full webhook name.

tldr: if a webhook's displayed name is `adam (he/him) | Demo System` and it knows that system's regular tag is `| Demo System`, it can use that to determine your servername is `adam (he/him)` (barring Unicode weirdness). that only works if it can rely on `| Demo System` to be unchanged, though. so i wouldn't hold my breath for a similar hack for servertags.
